### ISSUE TITLE
Values in PoolRealArray lose precission

### ISSUE THREAD

**Issue by lekoder** [godotengine/godot#45230]
**Godot version:**
3.2.3

**Issue description:**
When big float value is pushed into `PoolRealArray`, it looses precision.

**Steps to reproduce:**
```
	var v = 9631239965.144
	var pva = PoolRealArray()
	pva.append(v)
	print("float value:            %f"%v)
	print("value in PoolRealArray: %f" % pva[0])
	
	print("difference: %f" % [v - pva[0]])
```
**Minimal reproduction project:**
[GDScript Playground](https://gd.tumeo.space/KYDwLsB2AmDOAEA5A9tYAodBieBjANgIawIC2wpARsAE7wBuhNAloZfsAgObLwAWtYABp4AM2R1QhUgAcOALmwMm8QvAC88AExLGdShvgAiCOD1HMogK6Rc8APo1ghaAE8AFAEpFASD0NDAE4ANgBmAEYtUMCQgFYAOnCAFiT0PxUZRkMABWRkfAAlZ3wAQRoaQg9PNMzCeMIZGShod3pqnxkWSDB3I1F8ZEIwZXwrYHl4SanpyYBSUSNZtpqunqNGUeB4Zkh4XPyiwlLyyon5o3hZ+FqAbQAGAF12lZ216GZRUUFbccuFy-gN3o8AAtNdGPcHk80kA).

**Comment by lekoder** [comment]
It was pointed out to me that it can be actually a documentation problem. If that's the case and `PoolRealArray` is supposed to store 32-bit floats, it should be documented next to `PoolRealArray`. Right now it claims to:

> hold floating-point values ([float](https://docs.godotengine.org/pl/stable/classes/class_float.html#class-float))

...and there is no easy way to see that this will bite your *ss when you let the game run for extended period and your counters get high enough.

**Comment by akien-mga** [comment]
Yes that's a documentation issue, `PoolRealArray` is 32-bit and `PoolIntArray` too in 3.2.

In 4.0 you can choose `PackedFloat32Array` or `PackedFloat64Array`.

**Comment by aaronfranke** [comment]
As a workaround, you can use an untyped GDScript array to store 64-bit floats. However, since this stores Variants, note that the memory usage will be 24 bytes per item (20 bytes + padding) vs `PoolRealArray`'s 4 bytes per item (so a GDScript array will be 6x the memory usage, but in 4.0 `PackedFloat32Array` -> `PackedFloat64Array` would only increase memory usage by 2x).

**Comment by akien-mga** [comment]
Fixed by #45231.
